src/main/java/uk/gov/ons/ctp/common/rest/RestClient.java

```diff
@@ -174,43 +177,45 @@ public RestTemplate getRestTemplate() {
     HttpEntity<P> httpEntity = createHttpEntity(objToSend, headerParams);
     UriComponents uriComponents = createUriComponents(path, queryParams, pathParams);
     ResponseEntity<T> response;
+    String errorMessage = "request failed for the given path";
     try {
       response = restTemplate.exchange(uriComponents.toUri(), method, httpEntity, clazz);
     } catch (HttpStatusCodeException e) {
       // Failure detected. For 4xx and 5xx status codes
-      String errorMessage =
-          method.name()
-              + " request failed for path: '"
-              + uriComponents
-              + "' Status: "
-              + e.getStatusCode()
-              + " ResponseBody: '"
-              + e.getResponseBodyAsString()
-              + "'";
       if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-        log.warn(errorMessage);
+        logging
+            .with("path", path)
+            .with("methodName", method.name())
+            .with("uriComponents", uriComponents)
+            .with("Status", e.getStatusCode())
+            .with("ResponseBody", e.getResponseBodyAsString())
+            .warn(errorMessage);
       } else if (e.getStatusCode() == HttpStatus.TOO_MANY_REQUESTS) {
         // Caller expected to handle this situation
         log.info("Too many requests response on {} for path: {}", method.name(), path);
       } else {
-        log.error(errorMessage, e);
+        logging
+            .with("path", path)
```

Comment:
Why have we lost the body and the status here?

Reply:
The reason body & status is removed because we are showing the stacktrace which will include status and body anyway (.error(errorMessage, e)). If we add status and body here as well would be a duplication